Fix DeployStatus showing error when state is processing

martijngastkemper opened the pull request.

martijngastkemper committed:
Fix `isPublishing`

Not "ready" is the way to test for the site being live or publishing:
https://docs.netlify.com/api/get-started/#poll-for-deploy-state

martijngastkemper committed:
Use yarn in code-style workflow

martijngastkemper committed:
Upgrade actions

martijngastkemper committed:
Let Prettier ignore the dist directory

harmenjanssen commented on resources/js/components/DeployStatus.vue:
```diff
@@ -107,7 +107,7 @@ export default {
     },
     isPublishing(state) {
-      return state === "building" || state === "new" || state === "enqueued";
+      return state !== "ready";
```
Just checking, maar kan het voorkomen dat de state in error is, maar door deze aanpassing als _"is publishing"_ gezien wordt?

harmenjanssen reviewed (approved):
LGTM

martijngastkemper committed:
Fix state check for "error"

isPublished returned true when the state is error. By making explicit
functions it's hopefully less harder to mis states.

martijngastkemper commented:
You're right the error state wasn't handled properly. I've moved the state checks to separate functions now every state is covered. And I think it is easier to understand.

harmenjanssen reviewed (approved):
LGTM!
